extension/wallet/src/storage.rs

```diff
@@ -10,14 +11,11 @@ pub struct Storage {
 }
 
 impl Storage {
-    pub async fn get_open_loans(&self) -> Result<Vec<LoanDetails>> {
-        let loans = match self
-            .get_item::<String>("open_loans")
-            .context("no key \"open_loans\" in local storage")?
-        {
-            Some(loans) => serde_json::from_str(&loans)?,
-            None => Vec::<LoanDetails>::new(),
-        };
+    pub fn get_open_loans(&self) -> Result<Vec<LoanDetails>> {
+        let loans = self
+            .get_json_item::<Vec<LoanDetails>>("open_loans")
+            .context("no key 'open_loans' in local storage")?
+            .unwrap_or_default();
 
         Ok(loans)
     }
```

Comment:
A cleaner way of doing this would be to create an extension trait.

```rust
pub trait StorageExt {
	fn get_open_loans(&self) -> Result<Vec<LoanDetails>>;
}

impl StorageExt for Storage {
	
}
```

This way, you can have:

- `Storage` as a domain-independent convenience wrapper around `web_sys::Storage`
- convenience functions on `Storage` for loading / storing specific structs

However, given the new API, I am not sure if this function is even still worth having as it is basically a single expression, so you could even inline it and thereby reduce the indirection.

Reply:
I opted for removing it all together.